utils/clang-format-changed-only

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+set -u
```

Comment:
is this like clang-format-diff ?

Reply:
No - This checks the clang version to be used.

But instead of using the logic provided by franks awesome find command one could have come up with some regular expression for clang-format-diff.

Reply:
I have changed the script to use clang-format-diff